halo2_backend/src/plonk/circuit.rs

```diff
@@ -366,3 +366,117 @@ fn shuffle_argument_required_degree<F: Field, V: Variable>(arg: &shuffle::Argume
+
+        // simulate the expressios being used in a circuit
+        let l: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 0,
+            column_index: 0,
+            column_type: Any::Advice,
+            rotation: Rotation::cur(),
+        }));
+        let r: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 1,
+            column_index: 1,
+            column_type: Any::Advice,
+            rotation: Rotation::cur(),
+        }));
+        let o: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 2,
+            column_index: 2,
+            column_type: Any::Advice,
+            rotation: Rotation::cur(),
+        }));
+        let c: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 3,
+            column_index: 0,
+            column_type: Any::Fixed,
+            rotation: Rotation::cur(),
+        }));
+        let sl: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 4,
+            column_index: 1,
+            column_type: Any::Fixed,
+            rotation: Rotation::cur(),
+        }));
+        let sr: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 5,
+            column_index: 2,
+            column_type: Any::Fixed,
+            rotation: Rotation::cur(),
+        }));
+        let sm: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 6,
+            column_index: 3,
+            column_type: Any::Fixed,
+            rotation: Rotation::cur(),
+        }));
+        let so: ExpressionBack<Fr> = ExpressionBack::Var(VarBack::Query(QueryBack {
+            index: 7,
+            column_index: 4,
+            column_type: Any::Fixed,
+            rotation: Rotation::cur(),
+        }));
+
+        let simple_plonk_expr = sl * l.clone() + sr * r.clone() + sm * (l * r) - so * o + c;
+        assert_eq!(simple_plonk_expr.identifier(), "(((((Query(QueryBack { index: 4, column_index: 1, column_type: Fixed, rotation: Rotation(0) })*Query(QueryBack { index: 0, column_index: 0, column_type: Advice, rotation: Rotation(0) }))+(Query(QueryBack { index: 5, column_index: 2, column_type: Fixed, rotation: Rotation(0) })*Query(QueryBack { index: 1, column_index: 1, column_type: Advice, rotation: Rotation(0) })))+(Query(QueryBack { index: 6, column_index: 3, column_type: Fixed, rotation: Rotation(0) })*(Query(QueryBack { index: 0, column_index: 0, column_type: Advice, rotation: Rotation(0) })*Query(QueryBack { index: 1, column_index: 1, column_type: Advice, rotation: Rotation(0) }))))+(-(Query(QueryBack { index: 7, column_index: 4, column_type: Fixed, rotation: Rotation(0) })*Query(QueryBack { index: 2, column_index: 2, column_type: Advice, rotation: Rotation(0) }))))+Query(QueryBack { index: 3, column_index: 0, column_type: Fixed, rotation: Rotation(0) }))");
+    }
```

Comment:
Should we format this to allow for better reading at expense of more loc?

Reply:
Yes, I think it is better to improve the format.
Current `write_identifier` logic is simple - appending the identifiers at the end.
That's why we have such long, ugly identifier.
I will check the `write_identifier` if possible to improve.